boards/BMS/Test/Src/Test_StateMachine.cpp

```diff
@@ -60,6 +61,16 @@ FAKE_VALUE_FUNC(float, get_segment_3_voltage);
 FAKE_VALUE_FUNC(float, get_segment_4_voltage);
 FAKE_VALUE_FUNC(float, get_segment_5_voltage);
 
+FAKE_VALUE_FUNC(ExitCode, read_die_temperatures);
+FAKE_VALUE_FUNC(float, get_segment_0_die_temp_degc);
+FAKE_VALUE_FUNC(float, get_segment_1_die_temp_degc);
+FAKE_VALUE_FUNC(float, get_segment_2_die_temp_degc);
+FAKE_VALUE_FUNC(float, get_segment_3_die_temp_degc);
+FAKE_VALUE_FUNC(float, get_segment_4_die_temp_degc);
+FAKE_VALUE_FUNC(float, get_segment_5_die_temp_degc);
+FAKE_VALUE_FUNC(float, get_min_die_temp_degc);
+FAKE_VALUE_FUNC(float, get_max_die_temp_degc);
```

Comment:
You might want to consider adding a unit test where these functions change value and then check the CAN messages are updated as you'd expect.